index.d.ts

```diff
@@ -0,0 +1,177 @@
+export const GSS_C_DELEG_FLAG: number;
+export const GSS_C_MUTUAL_FLAG: number;
+export const GSS_C_REPLAY_FLAG: number;
+export const GSS_C_SEQUENCE_FLAG: number;
+export const GSS_C_CONF_FLAG: number;
+export const GSS_C_INTEG_FLAG: number;
+export const GSS_C_ANON_FLAG: number;
+export const GSS_C_PROT_READY_FLAG: number;
+export const GSS_C_TRANS_FLAG: number;
+
+export const GSS_C_NO_OID: number;
+export const GSS_MECH_OID_KRB5: number;
+export const GSS_MECH_OID_SPNEGO: number;
+
+/**
+ * Optional settings for *KerberosClient.wrap* method.
+ */
+export interface WrapOptions {
+    /**
+     * The user to authorize.
+     */
+    user?: string | undefined;
+
+    /**
+     * Indicates if the wrap should request message confidentiality.
+     */
+    protect?: boolean
+}
+
+/**
+ * Options for `initializeClient()`.
+ */
+export interface InitializeClientOptions {
+    /**
+     * Optional string containing the client principal in the form 'user@realm' (e.g. 'jdoe@example.com').
+     */
+    principal?: string | undefined;
```

Comment:
```suggestion
    principal?: string;
```

I mean I think that's the same? Maybe unless `exactOptionalPropertyTypes` is enabled?

Reply:
It is more limiting if `exactOptionalPropertyTypes` is enabled in the user's project, but this is how we type our other packages so I removed the `| undefined`s from this PR.